providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Essentials.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import org.jclouds.azurecompute.arm.util.GetEnumValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class Essentials {
+
+	public enum AlertState {
+		New, Acknowledged, Closed;
+
+		public static AlertState fromValue(final String text) {
+			return (AlertState) GetEnumValue.fromValueOrDefault(text, AlertState.New);
+		}
+	}
+
+	public enum MonitorCondition {
+		Fired, Resloved;
+
+		public static MonitorCondition fromValue(final String text) {
+			return (MonitorCondition) GetEnumValue.fromValueOrDefault(text, MonitorCondition.Fired);
+		}
+	}
+
+	@Nullable
+	public abstract ActionStatus actionStatus();
+
+	@Nullable
+	public abstract String alertRule();
+
+	@Nullable
+	public abstract String lastModifiedDateTime();
```

Comment:
`Date`?

Reply:
Use to keep as type as "String" instead of java.util.Date because of ISO-8601 format.
https://docs.microsoft.com/en-us/rest/api/monitor/alertsmanagement/alerts/get-all#essentials

Reply:
Modified as per review comment.